pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -262,3 +271,40 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *k
 	}
 	return nil
 }
+
+// cleanupOldDeployments triggers on deployment configuration update and add events in order
+// to delete old replication controller deployments if their quota has been reached
+func (c *DeploymentConfigController) cleanupOldDeployments(existingDeploymentList *kapi.ReplicationControllerList, deploymentConfig *deployapi.DeploymentConfig) error {
+	// we need to coerce the list type into something we can sort
+	existingDeployments := []*kapi.ReplicationController{}
+	for i := 0; i < len(existingDeploymentList.Items); i++ {
+		existingDeployments = append(existingDeployments, &existingDeploymentList.Items[i])
```

Comment:
All deployments associated with a dc should be in existingDeployments, see [here](https://github.com/openshift/origin/blob/d6879a68b89b0c0e92251ab7b7b8d0d4d47387bf/pkg/deploy/controller/deploymentconfig/controller.go#L74-L79) (except when we need to create the latest). There are two paths in the controller you need to distinguish. One is when [the dc version has no underlying replication controller](https://github.com/openshift/origin/blob/d6879a68b89b0c0e92251ab7b7b8d0d4d47387bf/pkg/deploy/controller/deploymentconfig/controller.go#L122-L139), which means we need to deploy one and the other is when we don't need to deploy a new one but [we need to reconcile the latest replication controller deployed](https://github.com/openshift/origin/blob/d6879a68b89b0c0e92251ab7b7b8d0d4d47387bf/pkg/deploy/controller/deploymentconfig/controller.go#L112-L121). In the former case, you should add the new deployment that the controller created to the list and then hand the list to your cleanup helper.